src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -1165,13 +1165,14 @@ void InboundLedger::runData ()
     {
         data.clear();
         {
-            ScopedLockType sl (mReceivedDataLock);
+            std::lock_guard<std::recursive_mutex> sl (mReceivedDataLock);
 
             if (mReceivedData.empty ())
             {
                 mReceiveDispatched = false;
                 break;
             }
+
             data.swap(mReceivedData);
         }
 
```

Comment:
👍  At first I didn't see why you didn't get rid of `ScopedLockType`, but I now see that would have a much larger ripple (no pun).